### PR TITLE
added generate config file

### PR DESCRIPTION
**Pull request by Stauber44**

### CODE REVIEW

**Review by HackedOS — CHANGES_REQUESTED** [review]
Dont directly hard code the config, instead create a config object and then serialize it to RON

**Comment by HackedOS** [comment]
this works perfectly fine but a hashmap is not ordered so the order in which the keybinds in the config are generated is random

File: src/config/mod.rs [hunk 1]
```diff
@@ -36,6 +46,98 @@ impl OutputConfig {
...
+            }
+        };
+
+        let mut keybinding_map = std::collections::HashMap::<KeyPattern, Action>::new();
+        keybinding_map.insert(
+            KeyPattern {
+                modifiers: KeyModifiersDef(vec![KeyModifier::Super]).into(),
+                key: xkb::KEY_Return,
+            },
+            Action::Spawn(String::from("alacritty")),
```
Comment (electron271) [inline src/config/mod.rs:76]: kitty is more universally used by default by window managers. use that

File: src/config/mod.rs [hunk 2]
```diff
@@ -36,6 +46,98 @@ impl OutputConfig {
     }
 }
 
+pub fn generate_config() -> PathBuf {
+    println!("Would you like to generate a config file? [y/N]");
```
[the remark this answers is not in the document]
Comment (HackedOS) [reply src/config/mod.rs:50]: i didnt test in tty lmao, let me test real quick

File: src/config/mod.rs (hunk 1 above)
Comment (HackedOS) [reply src/config/mod.rs:76]: nah

Comment (Insert5StarName) [reply src/config/mod.rs]: > kitty is more universally used by default by window managers. use that

Both ideas are stupid just use the default terminal / like i3 determine the installed one

**Review by HackedOS — APPROVED** [review]

**Pull request merged**